src/skydb.ts

```diff
@@ -100,6 +100,21 @@ export async function getJSON(
   // Lookup the registry entry.
   const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
   const { entry }: { entry: RegistryEntry | null } = await this.registry.getEntry(publicKey, dataKey, getEntryOpts);
+
+  // Calculate the new revision and get the cached revision.
+  const revision = entry?.revision ?? BigInt(-1);
+  const newRevision = incrementRevision(revision);
+  const cacheKey = getCacheKey(publicKey, dataKey);
+  const cachedRevision = this.revisionNumberCache[cacheKey];
+
+  if (cachedRevision && cachedRevision > newRevision) {
+    throw new Error("A higher revision number for this userID and path is already cached");
+  }
```

Comment:
1. Shouldn't this be `>=` instead of `>`? If we're trying to set revision 5 and that's already cached that means something is wrong.
2. If `cachedRevision` is 0 and we're trying to set 0 again this check won't be triggered.